node-graph/libraries/rendering/src/renderer.rs

```diff
@@ -1249,6 +1251,8 @@ impl Render for Table<Raster<CPU>> {
 				let id = row.source_node_id.map(|x| x.0).unwrap_or_else(|| {
 					let mut state = DefaultHasher::new();
 					image.data().hash(&mut state);
+					// Include document_id to prevent hash collisions between documents
```

Comment:
You can also trivially have collisions within the same document (e.g. by rotating a rectangle with only a thin stroke and a white fill).

I think therefore that it is not viable to hash the images here at all.